utils/log_parser.py

```diff
@@ -397,14 +526,22 @@ def process(logLines: List[str], dumpPhases: List[str]) -> None:
                     lastScopeName = lastScopeName.split("::")[-1]
                 assert lastScopeName == SCOPE_STACK[-1], "Exited scope must be same as the top of scope stack."
                 if (lastScopeName in dumpPhases):
-                    dump_dag("after_" + lastScopeName + "_" + str(phaseNo))
+                    dump_dag(
+                        f"after_{lastScopeName}_{phaseCounter}")
                 SCOPE_STACK.pop()
 
+                # Stop recording transformations.
+                if (lastScopeName == "optimizeFunctionBeforeLowering" or lastScopeName ==
```

Comment:
Why do you need to stop after these two specific transformations? Maybe define a set named "stopRecordTranformationName" containing these two names for now and just check `lastScopeName in stopRecordTranformationName`.

Reply:
The idea here is to know when the optimization entirely stops, which is normally when optimizeFunction or optimizeFunctionBeforeLowering exits. The log after the exit of these methods are mainly Function and Module destructor, which is not used in the determining the transformations.